apps/vaporgui/FlowEventRouter.cpp

```diff
@@ -1,137 +1,144 @@
+static RenderEventRouterRegistrar<FlowEventRouter> registrar(FlowEventRouter::GetClassType());
 
-
-FlowEventRouter::FlowEventRouter( QWidget *parent, ControlExec *ce) 
-                    : QTabWidget(parent),
-	                  RenderEventRouter( ce, FlowParams::GetClassType())
+FlowEventRouter::FlowEventRouter(QWidget *parent, ControlExec *ce)
+: RenderEventRouterGUI(ce, FlowParams::GetClassType())
 {
-	_variables = new FlowVariablesSubtab(this);
-	QScrollArea *qsvar = new QScrollArea(this);
-	qsvar->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
-	_variables->adjustSize();
-	qsvar->setWidget(_variables);
-	qsvar->setWidgetResizable(true);
-	addTab(qsvar, "Variables");
-
-	_seeding = new FlowSeedingSubtab(this);
-	QScrollArea* qsseed = new QScrollArea(this);
-	qsseed->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
-	qsseed->setWidget(_seeding);
-	qsseed->setWidgetResizable(true);
-    _seedingTab = qsseed;
-	addTab(qsseed,"Seeding");
-
-	_appearance = new FlowAppearanceSubtab(this);
-	QScrollArea* qsapp = new QScrollArea(this);
-	qsapp->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
-	qsapp->setWidget(_appearance);
-	qsapp->setWidgetResizable(true);
-	addTab(qsapp,"Appearance");
-
-	_geometry = new FlowGeometrySubtab(this);
-	QScrollArea *qsgeo = new QScrollArea(this);
-	qsgeo->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
-	qsgeo->setWidget(_geometry);
-	qsgeo->setWidgetResizable(true);
-	addTab(qsgeo, "Geometry");
+    AddSubtab("Variables", new PGroup({
+        new PSection("Variable Selection", {
+            new S::PDimensionSelector,
+            new S::PXFieldVariableSelector,
+            new S::PYFieldVariableSelector,
+            (new S::PZFieldVariableSelector)->OnlyShowForDim(3),
+            new S::PColorMapVariableSelector,
+        }),
+        new PFidelitySection
+    }));
+    
+    _seedingTab =
+    AddSubtab("Seeding", new PGroup({
+        new PSection("Flow Integration Settings", {
+            new PEnumDropdown(FP::_isSteadyTag, {"Streamlines", "Pathlines"}, {true, false}, "Flow Type"),
+            (new Pif(FP::_isSteadyTag))->Equals(true)->Then({
+                new PEnumDropdown(FP::_flowDirectionTag, {"Forward", "Backward", "Bi-Directional"}, {(int)FlowDir::FORWARD, (int)FlowDir::BACKWARD, (int)FlowDir::BI_DIR}, "Flow Direction"),
+                (new PIntegerSliderEdit(FP::_steadyNumOfStepsTag, "Integration Steps"))->SetRange(0, 5000),
+            })->Else({
+                _pathlineLengthSlider = new PIntegerSliderEdit(FP::_pastNumOfTimeSteps, "Pathline Length"),
+                _pathlineInjectionSlider = new PIntegerSliderEdit(FP::_seedInjInterval, "Injection Interval"),
+            }),
+            new PDoubleInput(FP::_velocityMultiplierTag, "Vector Field Multiplier"),
+            new PCheckbox(FP::_xPeriodicTag, "X axis periodicity"),
+            new PCheckbox(FP::_yPeriodicTag, "Y axis periodicity"),
+            new PCheckbox(FP::_zPeriodicTag, "Z axis periodicity"),
```

Comment:
I believe we are now setting default values in the Params constructors.  Correct?

Reply:
PWidgets try to avoid any notion of "defaults". Since they are required for the params API, all numbers are defaulted to 0 and all strings are empty.

Correct.